app/include/driver/i2c_master.h

```diff
@@ -53,22 +24,37 @@
 #define I2C_MASTER_SDA_LOW_SCL_LOW()  \
     gpio_output_set(0, 1<<I2C_MASTER_SDA_GPIO | 1<<I2C_MASTER_SCL_GPIO, 1<<I2C_MASTER_SDA_GPIO | 1<<I2C_MASTER_SCL_GPIO, 0)
 
-uint32 i2c_master_gpio_init(uint8 sda, uint8 scl, uint32 speed);
-void i2c_master_init(void);
-
 #define i2c_master_wait    os_delay_us
-void i2c_master_stop(void);
-void i2c_master_start(void);
-void i2c_master_setAck(uint8 level);
-uint8 i2c_master_getAck(void);
-uint8 i2c_master_readByte(void);
-void i2c_master_writeByte(uint8 wrdata);
-
-bool i2c_master_checkAck(void);
-void i2c_master_send_ack(void);
-void i2c_master_send_nack(void);
-
-uint8 i2c_master_get_pinSDA();
-uint8 i2c_master_get_pinSCL();
 
-#endif
+#endif //I2C_MASTER_OLD_VERSION
+
+typedef struct {
```

Comment:
Private to the C driver so why in the public header?

Reply:
My bad, fixed. I tempted to simplify "old version" by getting rid of macros like `I2C_MASTER_SDA_HIGH_SCL_HIGH` and such, but leaving `if-else` structure for replicating old timings, since there is no more binary equivalent of old code - only timings and behavior are simulated. It even supports multiple buses.

Reply:
:+1: